storage/remote/queue_manager.go

```diff
@@ -32,13 +33,25 @@ const (
 	subsystem = "remote_storage"
 	queue     = "queue"
 
-	defaultShards            = 10
+	// With a maximum of 500 shards, assuming an average of 100ms remote write
+	// time and 100 samples per batch, we will be able to push 500k samples/s.
+	defaultMaxShards         = 500
 	defaultMaxSamplesPerSend = 100
-	// The queue capacity is per shard.
-	defaultQueueCapacity     = 100 * 1024 / defaultShards
+
+	// defaultQueueCapacity is per shard - at 500 shards, this will buffer
+	// 50m samples.  It is configured to buffer 1024 batches, which at 100ms
+	// per batch is 1:40mins.
```

Comment:
The 1:40mins here is the buffer, used for blips and outages.  In normal operations, batches will have `defaultBatchSendDeadline` (5s) upper bound on the sends, and will tend towards an average of batch size / sample rate, which is almost always going to be much lower (1s-100ms).